### PR TITLE
Ncsdk 1036 dk button leds documentation

### PR DESCRIPTION
**Pull request by ru-fu**
Adding conceptual documentation for the DK Buttons and LEDs library, and reviewing the doxygen comments.

The rst file is currently in a different folder than the header file. @jhn-nordic Can we move the header file? And if so, there is probably more to do than just move the one file, or is that all that is needed?

### CODE REVIEW

**Comment by ru-fu** [comment]
Output on file://///hizz/components/ncs_docs/PR-384/nrf/include/dk_buttons_and_leds.html

**Comment by jhn-nordic** [comment]
> Adding conceptual documentation for the DK Buttons and LEDs library, and reviewing the doxygen comments.
> 
> The rst file is currently in a different folder than the header file. @jhn-nordic Can we move the header file? And if so, there is probably more to do than just move the one file, or is that all that is needed?

Looks good to me. I believe all you need to do is move the file and then remove "zephyr_include_directories(.)" from the CMakeLists.txt in the lib/dk_buttons_and_leds folder.

**Comment by jhn-nordic** [comment]
CI builds the asset tracker which is using this library. So it should be ok.

File: include/dk_buttons_and_leds.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+The DK Button and LED library is a simple module to interface with the buttons and LEDs on a Nordic Semiconductor development kit.
+It supports reading the state of up to four buttons or switches and controlling up to four LEDs.
+
+If you want to retrieve information about the button state, initialize the library with :cpp:func:`dk_buttons_init`.
+You can pass a callback function during initialization.
+This function is then called every time the button state changes.
+
+If you want to control the LEDs on the board, initialize the library with :cpp:func:`dk_leds_init`.
+You can then set the value of single LEDs, or set all of them to a state specified through bitmasks.
```
Comment (b-gent) [inline include/dk_buttons_and_leds.rst]:
``` suggestion
You can then set the value of a single LED, or set all of them to a state specified through bitmasks.
```

Comment (ru-fu) [reply include/dk_buttons_and_leds.rst]: ... now it's merged already. :( I'll change it in some other PR ...

**Pull request merged**